pkg/sgpp/utils.go

```diff
@@ -30,3 +38,68 @@ func LoadEnforcingModeFromEnv() EnforcingMode {
 		return DefaultEnforcingMode
 	}
 }
+
+func AddFeatureToCNINode(ctx context.Context, k8sClient client.Client, nodeName string, featureName rc.FeatureName, featureValue string) error {
```

Comment:
This function is not attached to any struct and is "dangling". Is it possible to attach this to a greater context, so we can avoid the complexity of passing dependencies around through method arguments (e.g. k8sClient). 

Another thought -- does it make sense to decouple these two concepts:
1. Constructs of the CNINode
2. Application of the CNINode

Something like:
```
node := v1.Node{}
_, err := client.Get(ctx, &node) { ... }
cniNode := CNINodeFor(node)
client.Patch(ctx, &cniNode) { ... }

Reply:
This can be added to ipamd.go instead of from another package. I prefer to create the CNINode objects from VPC resource controller side since in general VPC resource controller is persisting and tracking/managing the life time of the node object from its point of view. VPC CNI will restart for many reasons and seems no need to worry about creation of this CRD object, like it doesn't care the creation of node object.

Reply:
> I prefer to create the CNINode objects from VPC resource controller side since in general VPC resource controller is persisting and tracking/managing the life time of the node object from its point of view. 

Huge +1. I could see taking this a step further, and taking an assumption that the CRD exists for the given node before taking any actions as a CNI. WDYT?

Reply:
That would be nice. I am trying to find out if there is a k8s mechanism which is like a "generator" (the opposite side of finalizer) with Ownership. I think this is no such native controller to do this? In this case, we are relying on the VPC resource controller to generate the CRD object upon its owner node object's creation. Not sure if this is what you were referring to?

Reply:
Yeah there's no "generator" concept. I think of the CNINode is being owned by the CNI-agent, as it is the source of truth for the CNINode.spec. I'm not sure I see the benefit of creating the object from VPCRC, regardless of the restart strategy of VPCRC or VPCCNI.

What do you think about the following:
* VPCCNI wakes up, and applies (create/update) the CNINode to be what it expects it to be.
* VPCRC watches for the CNINode objects and takes action accordingly (i.e., attaching trunk eni)

Reply:
>the CNINode is being owned by the CNI-agent
I am not sure if this is correct since the agent will change for any reason but the CNINode should be living with the node. My preference is
- VPC RC (sort of) manage "nodes" and their owning CNINode objects
- VPC CNI doesn't have to worry about the CNINode creation since it should be taken care by VPC RC
- VPC CNI shouldn't have create permission on this resource
- VPC RC managing these objects will benefit future improvements (we can sync offline)

Reply:
Worth a discussion, I think.